up-l1/mqtt.adoc

```diff
@@ -35,8 +35,126 @@ MQTT is an OASIS standard messaging protocol for the Internet of Things (IoT). I
 
 For more information, please refer to https://mqtt.org/
 
-
 == Specifications
 
+=== UAttribute Mapping
+
+MQTT 5 supports custom header fields, and we leverage this to map the UAttribute values into the MQTT header.
+
+[cols="1,1"]
+|===
+| MQTT Header Property Code | Value (key:value)
+| UserProperty
+| ("1", "{UAttribute.id}") # UUID id
+| UserProperty
+| ("2", "{UAttribute.type}") # UMessageType
+| UserProperty
+| ("3","{UAttribute.source}") # Short Form Serialized Source UURI
+| UserProperty
+| ("4","{UAttribute.sink}") # Short Form Serialized Sink UURI
+| UserProperty
+| ("5","{UAttribute.priority}") # UPriority
+| UserProperty
+| ("6", "{UAttribute.ttl}") # Optional ttl
+| UserProperty
+| ("7", "{UAttribute.permissionLevel}") # Optional permission level
+| UserProperty
+| ("8", "{UAttribute.commStatus}") # Optional UCode commstatus
+| UserProperty
+| ("9", "{UAttribute.reqId}") # UUID reqid
+| UserProperty
+| ("10", "{UAttribute.token}") # Optional token
+| UserProperty
+| ("11", "{UAttribute.traceParent}") # Optional traceparent
+|===
+
+All uAttributes are mapped to a MQTT header UserProperty, where the key is the UAttribute protobuf field number.
+
+The source and sink fields **MUST** be link:../basics/uri.adoc[Short URI Format].
+
+The type field is used by a UListener to select message types on the topic.
+
+=== URI Mapping
+
+The URI mapping follows the short form UUri format, with the addition of a location identifier to determine if the message is intended for a local or cloud destination.
+
+- `{UUri.UAuthority.number}/{UUri.UEntity.id}/{UUri.UEntity.version_major}/{UUri.UResource.id}`
+
+  Note: If the UUri does not contain a UAuthority, add the UAuthority.number for the UAuthority the transport client is running on.
+
+Subscribe to all messages to the current Authority
+
+- `{UUri.UAuthority.number}/+/+/+` where `UAttributes.type != Publish`
+
+Subscribe to all messages from a different Authority
+
+- `{UUri.UAuthority.number}/+/+/+` where `UAttributes.type == Publish`
```

Comment:
If the subscribe operation has to use the UUri.uAuthority.number, it is not suitable for cloud to subscribe vehicle messages, given that the VIN is a 17-character string.

Reply:
@lynn2009 are you saying that the vin is too long for the topic name? I am unsure what other identifier to use, other than using the micro uri format for the topic name

Reply:
UAuthority.number is either id (VIN) or IP address.

Reply:
IMHO @lynn2009 refers to the fact that the cloud component that wants to receive all Publish messages from all vehicles would need to use a separate topic filter per subscribed topic/vehicle (`vin1/entity.id/1/resource.id`, `vin1/entity.id/1/resource.id`, etc) as opposed to only one filter (`pub.v1/+/+/+/+`) for Publish messages originating from any vehicle ...